Felienne/hedy
# [FEATURE] Automatically marks erroneous programs when sharing

**Description**
In this PR we add a new feature to keep track of erroneous shared programs. From now on programs are parsed before sharing and if an error occurs the user is presented with a confirmation modal. Then we store the parsing result (error : true / false) in the db as addition to the program stored. This way we can differentiate between correct and erroneous programs on the explore page.


**Fix for**
This PR fixes #1807 

**How to test**
This requires some testing as we have re-written some core functionality of our website. Login as any user and save either a correct or incorrect program. This process should work identical as before. When trying to share an incorrect program (either from the code page or the programs page) an additional pop-up is shown to verify that you are sure to share an erroneous program. Navigate to the explore page and notice that your faulty program now has a red header instead of the original blue one.
